File: app/src/main/java/igrek/songbook/dagger/DaggerIoc.java
package igrek.songbook.dagger;

import android.support.v7.app.AppCompatActivity;

import igrek.songbook.info.logger.Logger;
import igrek.songbook.info.logger.LoggerFactory;

public class DaggerIoc {
	
	private static FactoryComponent appComponent;
	
	private static Logger logger = LoggerFactory.getLogger();
	
	private DaggerIoc() {
	}
	
	public static void init(AppCompatActivity activity) {
		logger.info("Initializing Dagger IOC container...");
		appComponent = DaggerFactoryComponent.builder()
				.factoryModule(new FactoryModule(activity))
				.build();
	}
	
	public static FactoryComponent getFactoryComponent() {
		return appComponent;
	}
	
	/**
	 * only for testing purposes
	 */
	public static void setFactoryComponent(FactoryComponent component) {
		appComponent = component;
	}
}

File: app/src/main/java/igrek/songbook/dagger/DaggerIoc.kt
package igrek.songbook.dagger

import androidx.appcompat.app.AppCompatActivity
import igrek.songbook.info.logger.LoggerFactory

object DaggerIoc {

    lateinit var factoryComponent: FactoryComponent

    private val logger = LoggerFactory.logger

    private var factoryModule: FactoryModule? = null

    fun init(activity: AppCompatActivity) {
        logger.info("Initializing Dependencies container...")
        factoryModule = FactoryModule(activity)
        factoryComponent = DaggerFactoryComponent.builder()
                .factoryModule(factoryModule)
                .build()
    }
}
